### PR TITLE
chore: forbid unintentional exclusive tests

### PR DESCRIPTION
**Pull request by Lodin**
Addresses two points of #1760.

- Add an Eslint rule to forbid exclusive tests.
- Add a Mocha configuration that fail tests during CI if exclusive test is found.

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -16,6 +16,7 @@
     "import/prefer-default-export": "off",
     "max-params": "off",
     "max-classes-per-file": "off",
+    "mocha/no-exclusive-tests": "error",
```
Comment (sissbruecker) [inline .eslintrc:19]: I can still commit files that have been modified to use `it.only`. I think our lint-staged config is broken somehow. Though it seems to notice the issue in CI, which is good enough for me.

File: .eslintrc
Comment (sissbruecker) [inline .eslintrc:1]: This uncovered another usage of `it.only` in `packages/ts/generator-typescript-plugin-backbone/test/hierarchy-ref/HierarchyRef.spec.ts` 🙈. Could you please fix it?

File: .eslintrc (hunk 1 above)
Comment (Lodin) [reply .eslintrc:19]: Fixed that as well

**Review by sissbruecker — APPROVED** [review]
lint-staged works now 👍

**Pull request merged**